src/main/java/org/folio/dew/repository/FTPObjectStorageRepository.java

```diff
@@ -95,18 +103,18 @@ public void upload(String ftpUrl, String username, String password, String filen
     }
   }
 
-  private void changeWorkingDirectory(FTPClient ftpClient) throws IOException {
-    if (isDirectoryAbsent(ftpClient, ftpProperties.getWorkingDir())) {
-      log.info("A directory has been created: " + ftpProperties.getWorkingDir());
-      ftpClient.makeDirectory(ftpProperties.getWorkingDir());
+  private static void changeWorkingDirectory(FTPClient ftpClient, String path) throws IOException {
+    for (String dir : path.split("/")) {
```

Comment:
I am not sure that this separator will work both on Windows and Linux, maybe better to use constant like Files.separator?

Reply:
We have to escape it in the split so we can make code working for both systems but looks a bit ugly to me

Reply:
Found a better solution: `Pattern.quote`